<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">

    <ros2_control name="ros2_control_cybertruck" type="system">
      <hardware>
          <plugin>gz_ros2_control/GazeboSimSystem</plugin>
          <plugin>my_robot_hardware/MyRobotHardwareInterface</plugin>
          <param name="serial_port">/dev/ttyUSB0</param>
          <param name="baud_rate">115200</param>
      </hardware>

      <!-- Wheel Joints -->
      <joint name="joint_rotation_front_left">
        <command_interface name="velocity">
          <param name="min">-10</param>
          <param name="max">10</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="joint_rotation_front_right">
        <command_interface name="velocity">
          <param name="min">-10</param>
          <param name="max">10</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="joint_rotation_rear_left">
        <command_interface name="velocity">
          <param name="min">-10</param>
          <param name="max">10</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="joint_rotation_rear_right">
        <command_interface name="velocity">
          <param name="min">-10</param>
          <param name="max">10</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>

      <!-- Steering Joints -->
      <joint name="joint_steering_front_left">
        <command_interface name="position">
          <param name="min">-3.104</param>
          <param name="max">3.104</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="joint_steering_front_right">
        <command_interface name="position">
          <param name="min">-3.104</param>
          <param name="max">3.104</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="joint_steering_rear_left">
        <command_interface name="position">
          <param name="min">-3.104</param>
          <param name="max">3.104</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
      <joint name="joint_steering_rear_right">
        <command_interface name="position">
          <param name="min">-3.104</param>
          <param name="max">3.104</param>
        </command_interface>
        <state_interface name="position"/>
        <state_interface name="velocity"/>
      </joint>
    </ros2_control>
    
</robot>
